### PR TITLE
Fix clang build, freeing memory too early

### PR DESCRIPTION
**Pull request by Flakebi**
Only the top half of the last lap message was displayed when SuperTuxKart was compiled with clang (version 3.5.1), the bottom half was an empty string. @hiker found out that clang deletes the string too fast, so caching it as a local variable fixes that problem.

### CODE REVIEW

File: src/modes/linear_world.cpp [hunk 1]
```diff
@@ -350,10 +350,13 @@ void LinearWorld::newLap(unsigned int kart_index)
...
         std::string s = StringUtils::timeToString(time_per_lap);
 
-        irr::core::stringw m_fastest_lap_message;
+        // Store the temporary string because clang would mess this up
+        // (remove the stringw before the wchar_t* is used).
+        core::stringw css = kart->getName();
+
         //I18N: as in "fastest lap: 60 seconds by Wilber"
-        m_fastest_lap_message += _C("fastest_lap", "%s by %s", s.c_str(),
-                                    core::stringw(kart->getName()));
```
Comment (hiker) [inline src/modes/linear_world.cpp]: I noticed a few things we should have tried first:
1) can you try removing the core::stringw, i.e. just leaving
kart->getName()
2) if this doesn't work, what about kart->getName().c_str()
?

**Comment by Flakebi** [comment]
I improved it so it doesn't convert create a new stringw. Passing kart->getName() directly or even core::stringw(kart->getName().c_str()) doesn't help.

**Comment by hiker** [comment]
I tried to remove the unnecessary string copy when I applied this patch - please let me know if this should not work on clang, then I'll go back to the original patch.

Thanks a lot!
   Joerg

**Comment by Flakebi** [comment]
It works :-)

**Pull request merged**